### PR TITLE
Add docs for Hugging Face Auth provider

### PR DESCRIPTION
**Pull request by Nikpolik**
> [!IMPORTANT]
> 🔎 Previews:
>
> - https://clerk.com/docs/pr/1492/authentication/social-connections/huggingface

### This PR:

Adds docs for Hugging Face built in OAuth Provider. 

- [x] Add provider implementation - [PR](https://github.com/clerk/clerk_go/pull/7472).
- [x] Update `@clerk/types` with new constant - [PR](https://github.com/clerk/javascript/pull/4021)
- [x] Add provider in the dashboard - [PR](https://github.com/clerk/dashboard/pull/3495)
- [x] Docs - This PR

### CODE REVIEW

File: docs/authentication/social-connections/huggingface.mdx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+Finally copy the **Client ID** and **App secret** as shown in the image bellow. Go back to the Clerk Dashboard and paste them into the respective fields.
+
+<Images
+  width={1904}
+  height={981}
+  src="/docs/images/authentication-providers/huggingface/scopes-and-redirect-url.png"
+  alt="Creating a new application"
+/>
+
+Don't forget to click **Apply** in the Clerk dashboard. Congratulations! Social connection with Xero is now configured for your instance.
```
Comment (nikospapcom) [inline docs/authentication/social-connections/huggingface.mdx:56]:
```suggestion
Don't forget to click **Apply** in the Clerk dashboard. Congratulations! Social connection with Hugging Face is now configured for your instance.
```

Comment (Nikpolik) [reply docs/authentication/social-connections/huggingface.mdx:56]: Haha thank you for catching this!

File: docs/authentication/social-connections/huggingface.mdx [hunk 2]
```diff
@@ -0,0 +1,56 @@
+---
+title: Hugging Face
+description: Learn how to set up social connection with Hugging Face.
+---
+
+# Hugging Face
```
Comment (nikospapcom) [inline docs/authentication/social-connections/huggingface.mdx:6]: ✂️

Screenshot 2024-08-27 at 12 37 28 PM

File: docs/authentication/social-connections/huggingface.mdx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+First, you need to create a new Connected App. Navigate to **Settings > Connected Apps** and click on **Create application button**. On the following screen fill all the required information, such as the name, the logo etc.
+
+In the **Scopes** field its important to select at least _openid_, _profile_ and _email_.
+
+<Images
+  width={1904}
+  height={981}
+  src="/docs/images/authentication-providers/huggingface/create-application.png"
+  alt="Creating a new application"
+/>
```
Comment (nikospapcom) [inline docs/authentication/social-connections/huggingface.mdx:36]: I think you should replace `Images` component with 
```
![Creating a new application](/docs/images/authentication-providers/huggingface/create-application.png)
```

Comment (Nikpolik) [reply docs/authentication/social-connections/huggingface.mdx]: Yep updated thanks!

File: public/images/logos/auth_providers/huggingface.svg [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+</svg>
```
[the remark this answers is not in the document]
Comment (Nikpolik) [reply public/images/logos/auth_providers/huggingface.svg]: Fixed! Thank you for taking the time to notice this and improve it 🙏 It looks much better

**Review by kylemac — APPROVED** [review]

**Pull request merged**